.ci/tools/release-notes-generate.sh

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+set -e
+
+if [ -z "$1" ] || [ -z "$2" ]; then
+  echo "$0 <last version, e.g. 7.6.0> <new version, e.g. 7.7.0>"
+  exit 1
+fi
+
+BASEDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd)"
+LAST_VERSION="$1"
+RELEASE_VERSION="$2"
+
+CURL_API_HEADER=(--header "X-GitHub-Api-Version: 2022-11-28")
+CURL_AUTH_HEADER=()
+if [ -n "$GITHUB_TOKEN" ]; then
+  echo "Will use env var GITHUB_TOKEN for github REST API"
+  CURL_AUTH_HEADER=(--header "Authorization: Bearer $GITHUB_TOKEN")
+fi
+
+# determine current milestone
+MILESTONE_JSON=$(curl "${CURL_API_HEADER[@]}" "${CURL_AUTH_HEADER[@]}" -s "https://api.github.com/repos/pmd/pmd/milestones?state=all&direction=desc&per_page=10&page=1"|jq ".[] | select(.title == \"$RELEASE_VERSION\")")
+#DEBUG ONLY
+#MILESTONE_JSON='{"number":80,"closed_issues":40}'
+#DEBUG ONLY
+MILESTONE=$(echo "$MILESTONE_JSON" | jq .number)
+
+PAGE="1"
+HAS_NEXT="true"
+ISSUES_JSON=""
+while [ "$HAS_NEXT" = "true" ]; do
+    echo "Fetching issues for milestone ${MILESTONE} page ${PAGE}..."
+    URL="https://api.github.com/repos/pmd/pmd/issues?state=closed&sort=created&direction=asc&per_page=30&page=${PAGE}&milestone=${MILESTONE}"
+    RESPONSE="$(curl "${CURL_API_HEADER[@]}" "${CURL_AUTH_HEADER[@]}" -s -w "\n%header{link}" "$URL")"
```

Comment:
```suggestion
    RESPONSE="$(curl "${CURL_API_HEADER[@]}" "${CURL_AUTH_HEADER[@]}" -s -w "\nLink: %header{link}" "$URL")"
```

This makes sure, that the last line with the header is always written, even if there is no header (when the results fits into a single page and no paging is needed). In that case, the last line will be just `Link: `.
Then the following `tail` call for "LINK_HEADER" still works and retrieves exactly that line instead of cutting out a line from the body....